crates/rpc/rpc-engine-api/src/engine_api.rs

```diff
@@ -25,13 +26,15 @@ const MAX_PAYLOAD_BODIES_LIMIT: u64 = 1024;
 /// functions in the Execution layer that are crucial for the consensus process.
 pub struct EngineApi<Provider> {
     /// The provider to interact with the chain.
-    provider: Provider,
+    provider: Arc<Provider>,
```

Comment:
the assumption here is already that the `EngineApi` and hence the `Provider` are `Clone` so this isn't required.

what we should do here, what we already should have before, is to move all fields into a inner type that we then wrap in an arc

```rust
pub struct EngineApi<Provider> {
  inner: Arc<EngineApiInner<Provider>>
}

struct EngineApiInner<Provider> {
  // This is a now basically the same as EngineAPi 
}
```

all fields are very cheap to clone but they are cloned per request, so we can reduce this to one Arc::clone like this

Reply:
ok that makes sense! thanks for feedback